pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
 	if err != nil {
 		return err
 	}
-	return c.Do(request, 200, nil)
+	return c.Do(request, "200", nil)
+}
+
+//MCServerRolesInfo is the struct for the server roles of the localhost, in this case SplunkMonitoringConsole
+type MCServerRolesInfo struct {
+	ServerRoles []string `json:"server_roles"`
+}
+
+//MCDistributedPeers is the struct for information about distributed peers of the monitoring console
+type MCDistributedPeers struct {
+	ClusterLabel []string `json:"cluster_label"`
+	ServerRoles  []string `json:"server_roles"`
+}
+
+//AutomateMCApplyChanges change the state of new indexers from "New" to "Configured" and add them in monitoring console asset table
+func (c *SplunkClient) AutomateMCApplyChanges(mock bool) error {
+	if mock {
+		return nil
+	}
+	var configuredPeers, indexerMemberList, licenseMasterMemberList string
+	apiResponseServerRoles, err := c.GetMonitoringconsoleServerRoles()
+	if err != nil {
+		return err
+	}
+
+	apiResponseMCDistributedPeers := struct {
+		Entry []struct {
+			Name    string             `json:"name"`
+			Content MCDistributedPeers `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/services/search/distributed/peers"
+	err = c.Get(path, &apiResponseMCDistributedPeers)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range apiResponseMCDistributedPeers.Entry {
+		if configuredPeers == "" {
+			configuredPeers = e.Name
+		} else {
+			str := []string{configuredPeers, e.Name}
+			configuredPeers = strings.Join(str, ",")
+		}
+		for _, s := range e.Content.ServerRoles {
+			if s == "indexer" {
+				indexerMemberList = indexerMemberList + "&member=" + e.Name
+			}
+			if s == "license_master" {
+				licenseMasterMemberList = licenseMasterMemberList + "&member=" + e.Name
+			}
+		}
+	}
+
+	for _, e := range apiResponseServerRoles.ServerRoles {
+		if e == "indexer" {
+			indexerMemberList = "&member=localhost:localhost" + indexerMemberList
+		}
+		if e == "license_master" {
+			licenseMasterMemberList = licenseMasterMemberList + "&member=localhost:localhost"
+		}
+	}
+	reqBodyIndexer := indexerMemberList + "&default=true"
+	reqBodyLicenseMaster := licenseMasterMemberList + "&default=false"
+	err = c.UpdateDMCGroups("dmc_group_indexer", reqBodyIndexer)
```

Comment:
If possible now or later, it would be nice to mention the format of the full string which would be passed onto the end point. Not a necessity but would add to readability of code

Reply:
same above